dataline-api/build.gradle

```diff
@@ -15,41 +17,72 @@ openApiGenerate {
     modelPackage = "io.dataline.api.model"
 
     importMappings = [
-        'SourceSpecification' : 'com.fasterxml.jackson.databind.JsonNode',
-        'SourceConfiguration' : 'com.fasterxml.jackson.databind.JsonNode',
-        'DestinationSpecification' : 'com.fasterxml.jackson.databind.JsonNode',
-        'DestinationConfiguration' : 'com.fasterxml.jackson.databind.JsonNode',
+            'SourceSpecification'     : 'com.fasterxml.jackson.databind.JsonNode',
+            'SourceConfiguration'     : 'com.fasterxml.jackson.databind.JsonNode',
+            'DestinationSpecification': 'com.fasterxml.jackson.databind.JsonNode',
+            'DestinationConfiguration': 'com.fasterxml.jackson.databind.JsonNode',
+    ]
+
+    generateApiDocumentation = false
+
+    configOptions = [
+            dateLibrary  : "java8",
+            generatePom  : "false",
+            interfaceOnly: "true"
+    ]
+}
+
+task generateApiClient(type: GenerateTask) {
+    generatorName = "java"
+    inputSpec = specFile
+    outputDir = "$buildDir/generated"
+
+    apiPackage = "io.dataline.api.client"
+    invokerPackage = "io.dataline.api.client.invoker"
+    modelPackage = "io.dataline.api.client.model"
+
+    importMappings = [
+            'SourceSpecification'     : 'com.fasterxml.jackson.databind.JsonNode',
+            'SourceConfiguration'     : 'com.fasterxml.jackson.databind.JsonNode',
+            'DestinationSpecification': 'com.fasterxml.jackson.databind.JsonNode',
+            'DestinationConfiguration': 'com.fasterxml.jackson.databind.JsonNode',
     ]
 
+    library = "native"
+
     generateApiDocumentation = false
 
     configOptions = [
             dateLibrary  : "java8",
             generatePom  : "false",
             interfaceOnly: "true"
     ]
+
 }
 
 dependencies {
     implementation group: 'com.fasterxml.jackson.core', name: 'jackson-annotations'
     implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind'
+    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310'
 
     implementation group: 'io.swagger', name: 'swagger-annotations', version: '1.6.2'
 
     implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
     implementation group: 'javax.ws.rs', name: 'javax.ws.rs-api', version: '2.1.1'
     implementation group: 'javax.validation', name: 'validation-api', version: '2.0.1.Final'
+
+    implementation group: 'org.openapitools', name: 'jackson-databind-nullable', version: '0.2.1'
 }
 
 sourceSets {
     main {
         java {
-            srcDir "$buildDir/generated/src/gen/java"
+            srcDirs "$buildDir/generated/src/gen/java", "$buildDir/generated/src/main/java", "$projectDir/src/main/java"
```

Comment:
can you put both the client and the sever code in the same generated/gen directory?

Reply:
this is generator specific unfortunately. The `java` generator puts them in the `main` directory